### PR TITLE
Task/rider clean architecture

### PR DESCRIPTION
**Pull request by AhmedGhayada23**
install packages :
  flutter_screenutil: ^5.5.4
  get: ^4.6.5
  dio: ^4.0.6
  get_storage: ^2.0.3

add localization arabic  and english .
add Themes dark and light .
add classes colors and images and sizes .
add route class contains the name of pages and list of GetPages .
Use a library screenutil .

### CODE REVIEW

File: lib/features/home/presentation/pages/home_page.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            Text('Select_language'.tr),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: [
+                ElevatedButton(
+                    onPressed: () => myLocaleController.changeLanguage('en'),
+                    child: const Text('English')),
```
Comment (fahjan) [inline lib/features/home/presentation/pages/home_page.dart:26]: localize it

File: lib/features/home/presentation/pages/home_page.dart [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                ElevatedButton(
+                    onPressed: () => myLocaleController.changeLanguage('ar'),
+                    child: const Text('عربي')),
+              ],
```
Comment (fahjan) [inline lib/features/home/presentation/pages/home_page.dart:30]: localized it please

File: lib/main.dart [hunk 3]
```diff
@@ -7,109 +12,21 @@ void main() {
...
+        minTextAdapt: true,
+        splitScreenMode: true,
+        builder: (context, child) {
+          return GetMaterialApp(
+            translations: LocaleString(),
+            debugShowCheckedModeBanner: false,
+            initialRoute: AppRoutes.home,
+            getPages: pages,
+            locale: const Locale('en', 'US'),
+            home: Container(),
```
Comment (fahjan) [inline lib/main.dart:28]: no need for home

File: lib/utils/routing/pages.dart [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import 'package:gtc_rider/features/home/presentation/binding/home_binding.dart';
+import 'package:gtc_rider/features/home/presentation/pages/home_page.dart';
+import 'package:gtc_rider/utils/routing/app_routes.dart';
+
+final List<GetPage> pages = <GetPage>[
+  GetPage(
+    name: AppRoutes.home,
+    page: () => const HomePage(),
+    binding: HomeBinding(),
+    // fullscreenDialog: false,
```
Comment (fahjan) [inline lib/utils/routing/pages.dart:11]: no need for commented code.
or comment it as TODO
// TODO(mohammed-fahjan): To do things

File: pubspec.lock [hunk 5]
```diff
@@ -50,13 +50,34 @@ packages:
       url: "https://pub.dartlang.org"
```
Comment (fahjan) [inline pubspec.lock:50]: ignore this file

File: pubspec.yaml [hunk 6]
```diff
@@ -34,6 +34,11 @@ dependencies:
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.2
+  flutter_screenutil: ^5.5.4
+  get: ^4.6.5
+  dio: ^4.0.6
+  get_storage: ^2.0.3
+
```
Comment (fahjan) [inline pubspec.yaml:41]: install onesignal

File: lib/features/home/presentation/pages/home_page.dart (hunk 1 above)
Comment (AhmedGhayada23) [reply lib/features/home/presentation/pages/home_page.dart]: solve problem

File: lib/features/home/presentation/pages/home_page.dart (hunk 2 above)
Comment (AhmedGhayada23) [reply lib/features/home/presentation/pages/home_page.dart:30]: solve problem

File: lib/main.dart (hunk 3 above)
Comment (AhmedGhayada23) [reply lib/main.dart]: solve problem

File: lib/utils/routing/pages.dart (hunk 4 above)
Comment (AhmedGhayada23) [reply lib/utils/routing/pages.dart]: solve problem

File: pubspec.lock (hunk 5 above)
Comment (AhmedGhayada23) [reply pubspec.lock]: solve problem

**Review by mohammedalian — APPROVED** [review]

**Pull request merged**